### DISCUSSION TITLE
HLS Preview Generator requests Auth, no Auth mentioned in documentation

### DISCUSSION THREAD

**Discussion by fsaavedrap** [EyevinnOSC/community#26] [Q&A] [answered]
Hello, I'm trying to test the HLS Preview Generator. I have subscribed and launched a service. I'm sending a GET request as shown in the documentation, but I'm getting a 401 error:

<html>
<head><title>401 Authorization Required</title></head>
<body>
<center><h1>401 Authorization Required</h1></center>
<hr><center>nginx</center>
</body>
</html>

There is no mention of authentication or credentials in the documentation for the service, what am I missing?

**Comment by birme** [comment, marked as answer]
Hi @fsaavedrap
Thank you for posting the question!

The endpoints `/image` and `/video` should not require any authorization. To verify I created an instance and requested a preview image with the following URL:
```
https://eyevinn-test.eyevinn-preview-hls-service.auto.prod.osaas.io/image?u=http%3A%2F%2Fmaitv-vod.lab.eyevinn.technology%2FVINN.mp4%2Fmaster.m3u8
```

Best regards,
   Jonas

**Reply by fsaavedrap** [reply]
I've managed to get it to throw out a 502 now, with this request:

`curl --location 'https://ndmedawrks-clipharvester2.eyevinn-preview-hls-service.auto.prod.osaas.io/image?u=https%3A%2F%2Fdalosdnhj8b1.cloudfront.net%2Fgrabacion-ATV%2Fplaylist.m3u8'`

**Reply by birme** [reply]
I see, the documentation should be more clear on this, but the service only supports VOD and not a live stream.

**Comment by fsaavedrap** [comment]
ahhh, ok. Is there another service on the platform that should be used for
live?

**Reply by birme** [reply]
I am checking the code in the open source repository for this one, and now I am wondering whether the issue is that you are referring to a media playlist URL and not the "master" manifest. Do you have that one and could try with that instead?

**Comment by fsaavedrap** [comment]
That works! The encoder was not creating a master manifest. I made one by
hand to call the same manifest and now it works. Thanks a lot for the help.

**Reply by birme** [reply]
I am happy to hear that. Glad I could help!

**Comment by fsaavedrap** [comment]
Follow up question: I see that the previews generated are always 4s long,
would it be possible to pass a duration in the query to change that?

**Reply by birme** [reply]
The preview is the duration of one segment in the media playlist. In the case of VOD you choose another position / segment with the pos query parameter but duration cannot be changed
